### PR TITLE
Showcase usage of frozen for partial function memoization

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@cbeck88  | @jaydee-io I'll appreciate a small review / some comments on that use case if you have time :-)

**Comment by jaydee-io** [comment]
:+1:  Nice use case of a '_memoization with compile-time computed cached values_'

File: examples/partial_memoization.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        if(where != cached.end()) {
+            return where->second;
+        }
+        return F{}(v);
+    }
+};
+
+#include <iostream>
+int main(int argc, char** argv) {
+  memoized<prime_sieve, uint64_t, 0,1,2,3,4,5,6,7,8,9> ps;
```
[the remark this answers is not in the document]
Comment (serge-sans-paille) [reply examples/partial_memoization.cpp]: Yeah that was my initial thought. But we could also accept a single oject that holds an integer sequence.

**Comment by serge-sans-paille** [comment]
I fixed the obvious problem on the code sample, leaving the remaining like this and merging. Example presented at cppcon o/

File: examples/partial_memoization.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        static constexpr frozen::map<T, decltype(F{}(v)), sizeof...(Vs)> cached = {{Vs, F{}(Vs)}...};
+        auto where = cached.find(v);
+        if(where != cached.end()) {
+            return where->second;
+        }
+        return F{}(v);
+    }
+};
+
+#include <iostream>
```
[the remark this answers is not in the document]
Comment (serge-sans-paille) [reply examples/partial_memoization.cpp]: The goal was to not clutter the actual usage with the calling code ;-)

**Comment by serge-sans-paille** [comment]
Thanks @jaydee-io for the review. You get the cite you deserved at cppon! And so did @cbeck88 and @apmccartney, obviously.

**Pull request closed without merging**